utils/terrascript_client.py

```diff
@@ -1726,3 +1735,183 @@ def unpack_namespace_info(self, namespace_info):
+            raise ElasticSearchResourceNameInvalidError(
+                f"[{account}] ElasticSearch domain name must must start with" +
+                f" a lowercase letter and must be between 3 and 28 " +
+                f"characters. Valid characters are a-z (lowercase only), 0-9" +
+                f", and - (hyphen). " +
+                f"{values['identifier']}")
+
+        # values['domain_name'] = identifier
+        elasticsearch_version = values.get('elasticsearch_version')
+        if not self.validate_elasticsearch_version(elasticsearch_version):
+            raise ElasticSearchResourceVersionInvalidError(
+                f"[{account}] Invalid ElasticSearch version" +
+                f" {values['elasticsearch_version']} provided" +
+                f" for resource {values['identifier']}.")
+
+        values['cluster_config']["zone_awareness_enabled"] = False
+
+        es_values = {}
+        es_values["domain_name"] = identifier
+        es_values["elasticsearch_version"] = \
+            values.pop('elasticsearch_version')
+
+        ebs_options = values.get('ebs_options', {})
+
+        es_values["ebs_options"] = {
+            "ebs_enabled": ebs_options.get('ebs_enabled', True),
+            "volume_size": ebs_options.get('volume_size', '100')
+        }
+
+        es_values["encrypt_at_rest"] = {
+            "enabled": values.get('encrypt_at_rest', {}).get('enabled', True)
+        }
+
+        node_to_node_encryption = values.get('node_to_node_encryption', {})
+
+        es_values["node_to_node_encryption"] = {
+            "enabled": node_to_node_encryption.get("enabled", True)
+        }
+
+        domain_endpoint_options = values.get('domain_endpoint_options', {})
+        tls_security_policy = \
+            domain_endpoint_options.get("tls_security_policy",
+                                        'Policy-Min-TLS-1-0-2019-07')
+        enforce_https = domain_endpoint_options.get("enforce_https", True)
+        es_values["domain_endpoint_options"] = {
+            "enforce_https": enforce_https,
+            "tls_security_policy": tls_security_policy
+        }
+
+        cluster_config = values.get('cluster_config', {})
+        dedicated_master_enabled = cluster_config.\
+            get("dedicated_master_enabled", True)
+        dedicated_master_type = cluster_config.\
+            get("dedicated_master_type", 't2.small.elasticsearch')
+        dedicated_master_count = cluster_config.\
+            get("dedicated_master_count", 3)
+        zone_awareness_enabled = cluster_config.\
+            get("zone_awareness_enabled", True)
+
+        es_values["cluster_config"] = {
+            "instance_type": cluster_config.get("instance_type",
+                                                't2.small.elasticsearch'),
+            "instance_count": cluster_config.get("instance_count", 3),
+            "dedicated_master_enabled": dedicated_master_enabled,
```

Comment:
If dedicated master is disabled, will tf error or create bad resource if master_type/count are set?

Reply:
Now I only add dedicated master config if enabled